src/bootstrap/compile.rs

```diff
@@ -344,8 +344,8 @@ impl Step for StartupObjects {
         t!(fs::create_dir_all(dst_dir));
 
         for file in &["rsbegin", "rsend"] {
-            let src_file = &src_dir.join(file.to_string() + ".rs");
-            let dst_file = &dst_dir.join(file.to_string() + ".o");
+            let src_file = &src_dir.join((*file).to_string() + ".rs");
+            let dst_file = &dst_dir.join((*file).to_string() + ".o");
```

Comment:
Please drop these changes as they hurt readability. Maybe a clippy bug?

Reply:
The [inefficient_to_string](https://rust-lang.github.io/rust-clippy/master/index.html#inefficient_to_string) lint fired here